### PR TITLE
Update loss function and added model.eval() mode

### PR DESCRIPTION
**Pull request by avinashsai**
1. log_softmax is an efficient activation function compared to softmax because it avoids overflow when 
    there are large numbers. Hence updated loss function accordingly.
2. It is always recommended to use .eval() mode while testing especially when Dropout layers are used. 
    Hence added model.eval() mode accordingly.
3. While validation it is better to use torch.no_grad() as it lowers the memory by setting autograd to False.

### CODE REVIEW

**Comment by rasbt** [comment]
Thanks, this is old code and I agree with all of that! :)

> It is always recommended to use .eval()

In this particular code it may not make a difference because I neither use batchnorm nor dropout. Nonetheless, it's a good thing to add.

> While validation it is better to use torch.no_grad() as it lowers the memory by setting autograd to False.

I usually use 

    with torch.set_grad_enabled(False):
        ...

Is there are difference between the two?

**Comment by avinashsai** [comment]
There is one minor difference torch.set_grad_enabled(False) can turn off/on gradients where as torch.no_grad() sets only to off

**Comment by rasbt** [comment]
ah ok, so this is also a shorthand then. Was thinking that one of the two was deprecated but I was probably confusing it with the old `volatile`. LGTM, thanks!

**Comment by rasbt** [comment]
sure. Could you please insert a new cell at the bottom as the last cell though to display the software versions used: 

%load_ext watermark
%watermark -iv

(please also use PyTorch 1.0 for the updates)

**Comment by avinashsai** [comment]
Sure I will do the updates for 1.0 and will add the cell at the bottom.
